lib/manageiq/api/common/application_controller_mixins/exception_handling.rb

```diff
@@ -0,0 +1,37 @@
+module ManageIQ
+  module API
+    module Common
+      module ApplicationControllerMixins
+        module ExceptionHandling
+          def self.included(other)
+            other.rescue_from(StandardError, RuntimeError) do |exception|
+              errors, status = collected_ipp_errors(exception)
+
+              render :json => errors, :status => status
+            end
+          end
+
+          def collected_ipp_errors(exception, default_code = 400)
```

Comment:
I'm assuming `ipp_` stands for the Insights platform proposal? Maybe just `collected_errors` since that better defines the method behavior.